tests/integration/test_features_api.py

```diff
@@ -0,0 +1,276 @@
+# Copyright 2025 Planet Labs PBC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+#
+from http import HTTPStatus
+import json
+from typing import Any
+import httpx
+import pytest
+import respx
+
+from planet import FeaturesClient, Session
+from planet.sync.features import FeaturesAPI
+
+pytestmark = pytest.mark.anyio  # noqa
+
+# Simulated host/path for testing purposes. Not a real subdomain.
+TEST_URL = 'http://test.planet.com/features/v1/ogc/my'
+
+TEST_GEOM = {"type": "Polygon", "coordinates": [[[]]]}
+
+TEST_FEAT = {"type": "Feature", "geometry": TEST_GEOM}
+
+TEST_COLLECTION_1 = {
+    "id": "collection1",
+    "title": "Collection 1",
+    "description": "test collection 1"
+}
+
+TEST_COLLECTION_2 = {
+    "id": "collection2",
+    "title": "Collection 2",
+    "description": "test collection 2"
+}
+
+TEST_COLLECTION_LIST = [TEST_COLLECTION_1, TEST_COLLECTION_2]
+
+
+class ExampleGeoInterface:
+
+    @property
+    def __geo_interface__(self) -> dict:
+        return TEST_GEOM
+
+
+def mock_response(url: str, json: Any, method: str = 'get'):
+    mock_resp = httpx.Response(HTTPStatus.OK, json=json)
+    respx.request(method, url).return_value = mock_resp
+
+
+def to_collection_model(collection: dict) -> dict:
+    """create a collection model as it would appear in a Features API response"""
+    id = collection.get("id")
+    return {
```

Comment:
less on you and more on our linter config but I find this to be unreadable... we should figure this out and reformat everything to avoid conflict/churn

Reply:
updated